projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -926,7 +926,9 @@ export class IgxDatePickerComponent extends PickerBaseDirective implements Contr
             }
             // do not focus the input if clicking outside in dropdown mode
             if (this.getEditElement() && !(args.event && this.isDropdown)) {
-                this.inputDirective.focus();
+                // this avoids the grid receiving a focus out because the overlay is removed in the same execution
+                // chain causing the browser to focus the body element first
+                setTimeout(() => this.inputDirective.focus());
```

Comment:
Kinda sus. The grid receives multiple focus out events, _most_ of which are from in-between focus shifting to the document body. Pretty much every cell click does that and that's why the checks on the handler are so strict about what we want to handle.

Reply:
Okay, the cell click traced down to `‎IgxCellCrudState.updateCell`:
https://github.com/IgniteUI/igniteui-angular/blob/d6e7b0e74fd51fa7bb5fb65db309c6197d721121/projects/igniteui-angular/src/lib/grids/common/crud.service.ts#L259-L263

That blur will force the focus to jump temporarily outside the grid and return, tricking the app scenario usage. This IIRC can be considered a minor bug on the Grid side and it might be better if we avoid blanket blur in general. If replaced with `this.grid.tbody.nativeElement.focus()` instead it should work fine, though the fix for #14020 should be re-validated.

Reply:
As for the Date Picker - that's another component issue. The entire KB nav of the date picker is broken:
https://www.infragistics.com/products/ignite-ui-angular/angular/components/date-picker#angular-datepicker-example Pick a value, focus _does not_ return to the input.